LLDP@VDSM

EdDev opened the pull request.

EdDev committed:
LLDP@VDSM

dankenigsberg commented on design/network/lldp/lldp.md:
```diff
@@ -4,3 +4,47 @@
+
+## Requirements
+
+- Expose through oVirt REST the LLDP TLV information from the host interfaces.
```
Expose LLDP TLV information of host interfaces through oVirt REST

EdDev replied:
Done

dankenigsberg commented on design/network/lldp/lldp.md:
```diff
@@ -4,3 +4,47 @@
+- Expose through oVirt REST the LLDP TLV information from the host interfaces.
+
+- Requirements open questions:
+  - At what resolution the LLDP info request is to be supported? One request per cluster/host/iface/tlv?
```
REST is not yet defined, but I think that when requesting all of the details of a specific HostNIC, the REST client should get a dictionary of all knowns TLVs. This fits our current design, and seems to be simple to use by our customers

EdDev replied:
That assumes that the customer will render the json/xml to extract the information. Perhaps they expect something specific, I suggest asking them.
BTW: If we are to provide the info just per NIC, then caching the LLDP data as one big chunk is a valid option.

It's also a question of how the info is fetched: Should it be fetched using the all mighty caps report or we need a dedicated finer grained verb to fetch this info.

dankenigsberg commented on design/network/lldp/lldp.md:
```diff
@@ -4,3 +4,47 @@
+
+- Requirements open questions:
+  - At what resolution the LLDP info request is to be supported? One request per cluster/host/iface/tlv?
+  - What is the required resolution of the information in relation to time (caching)? If this info is considered 'real time', then no caching is to be applied, the data is valid only when quieried. If the info is to be cached, at what interval it needs to be asked?
```
I think that making a REST call trigger a synchronous Vdsm call does not fit very well into Engine's design. Usually, Engine polls Vdsm, stores the (even volatile) data in DB, and exposes it on demand to UI and REST.

dominikholler replied:
It would be consistent if the LLDP information are as old as the other dynamic information about the host.

EdDev replied:
I understand you points, but consider that:
- The user cannot trust the information if it is not current, I think that the cached data should be invalidated after a defined period.
- "The other dynamic information" is a too broad, each info has its own resolution.

We still need to define the time resolution of this info explicitly and say something like: This info is refreshed every 30sec, or, this info is true as of the last capability report (and give the time perhaps).

dominikholler replied:
From my point of view this is more relaxed, since I expect that the update frequency of this information is very low. If the user wants to enforce refreshed data, he could refresh the informations by triggering an capability report.

dankenigsberg commented on design/network/lldp/lldp.md:
```diff
@@ -4,3 +4,47 @@
+
+## VDSM implementation options
+
+VDSM is planned to collect LLDP information through LLDPDA or NetworkManager services.
```
LLDPDA? `lldpad` maybe?

EdDev replied:
Done

dankenigsberg commented on design/network/lldp/lldp.md:
add a link to patch

EdDev replied:
Done

EdDev committed:
lldptool report output example

EdDev committed:
Refrase requirement based on review

EdDev committed:
Spelling fix

EdDev committed:
Add link to the related VDSM pathces.

EdDev committed:
NetworkManager LLDP report example

Output for the following command:
nmcli -f all device lldp list ifname enp5s0f0

dominikholler commented on design/network/lldp/lldp.md:
```diff
@@ -17,7 +17,7 @@ This information is usefull for operators and admins to manage and troubleshoot
 
 ## VDSM implementation options
```
Would be systemd-networkd an implementation option, too?

EdDev replied:
I do not think so, we will use NetworkManager. We have no plans for networkd at the moment.

dominikholler commented on design/network/lldp/lldp.md:
```diff
@@ -4,3 +4,110 @@
+## VDSM implementation options
+
+VDSM is planned to collect LLDP information through LLDPAD or NetworkManager services.
+- LLDPDA is as imidiate silution to the data collection as it can be used by VDSM as a driver without any special work dependency.
```
We should ensure, that https://github.com/jrfastab/lldpad/blob/48a5f38778b18d6659a672ccb4640f25c6720827/lldptool_cmds.c#L451 is not dangerous regarding long values.

dominikholler commented on design/network/lldp/lldp.md:
```diff
@@ -4,3 +4,110 @@
+  - Should we support all possible TLV entires or limit/filter specific ones.
+
+## VDSM implementation options
+
```
https://github.com/openstack/ironic-inspector should mentioned here.
